### PR TITLE
SONARJAVA-4988: Use SonarLintCache component and make it accessible to custom rules via the caching APIs

### PR DESCRIPTION
**Pull request by anton-haubner-sonarsource**
Please ensure your pull request adheres to the following guidelines: 

- [ ] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [ ] Unit tests are passing and you provided a unit test for your fix
- [ ] ITs should pass : To run ITs locally, checkout the [README](https://github.com/SonarSource/sonar-java/blob/master/README.md) of the project.
- [ ] If there is a [Jira](http://jira.sonarsource.com/browse/SONARJAVA) ticket available, please make your commits and pull request start with the ticket number (SONARJAVA-XXXX)

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/java/SonarComponents.java [hunk 1]
```diff
@@ -341,7 +379,8 @@ void reportIssue(AnalyzerMessage analyzerMessage, RuleKey key, InputComponent fi
       if (!textSpan.onLine()) {
         Preconditions.checkState(!textSpan.isEmpty(), "Issue location should not be empty");
       }
-      issue.setPrimaryLocation((InputFile) fileOrProject, analyzerMessage.getMessage(), textSpan.startLine, textSpan.startCharacter, textSpan.endLine, textSpan.endCharacter);
+      issue.setPrimaryLocation((InputFile) fileOrProject, analyzerMessage.getMessage(), textSpan.startLine, textSpan.startCharacter,
+        textSpan.endLine, textSpan.endCharacter);
```
Comment (anton-haubner-sonarsource) [inline java-frontend/src/main/java/org/sonar/java/SonarComponents.java:383]: Nothing changed here, this is just the auto formatting.

File: java-frontend/src/main/java/org/sonar/java/SonarComponents.java [hunk 2]
```diff
@@ -513,7 +552,8 @@ public boolean fileCanBeSkipped(InputFile inputFile) {
     } catch (ApiMismatchException e) {
       if (!alreadyLoggedSkipStatus) {
         LOG.info(
-          "Cannot determine whether the context allows skipping unchanged files: canSkipUnchangedFiles not part of sonar-plugin-api. Not skipping. {}",
+          "Cannot determine whether the context allows skipping unchanged files: canSkipUnchangedFiles not part of sonar-plugin-api. " +
+            "Not skipping. {}",
```
Comment (anton-haubner-sonarsource) [inline java-frontend/src/main/java/org/sonar/java/SonarComponents.java:556]: Again, this is just a formatting change.

File: java-frontend/src/test/java/org/sonar/java/SonarComponentsTest.java [hunk 3]
```diff
@@ -115,10 +116,14 @@ class SonarComponentsTest {
...
-  private static final String LOG_MESSAGE_FILES_CANNOT_BE_SKIPPED = "The Java analyzer cannot skip unchanged files in this context. A full analysis is performed for all files.";
-  private static final String LOG_MESSAGE_CANNOT_DETERMINE_IF_FILES_CAN_BE_SKIPPED = "Cannot determine whether the context allows skipping unchanged files: canSkipUnchangedFiles not part of sonar-plugin-api. Not skipping. {}";
+  private static final String LOG_MESSAGE_FILES_CAN_BE_SKIPPED =
+    "The Java analyzer is running in a context where unchanged files can be skipped. " +
+      "Full analysis is performed for changed files, optimized analysis for unchanged files.";
+  private static final String LOG_MESSAGE_FILES_CANNOT_BE_SKIPPED =
+    "The Java analyzer cannot skip unchanged files in this context. A full analysis is performed for all files.";
+  private static final String LOG_MESSAGE_CANNOT_DETERMINE_IF_FILES_CAN_BE_SKIPPED =
+    "Cannot determine whether the context allows skipping unchanged files: " +
+      "canSkipUnchangedFiles not part of sonar-plugin-api. Not skipping. {}";
```
Comment (anton-haubner-sonarsource) [inline java-frontend/src/test/java/org/sonar/java/SonarComponentsTest.java:126]: These are again just formatting changes.

**Review by alban-auzeill — APPROVED** [review]
LGTM, good work in the comments to explain why we have this temporary hack, it should help later to remove it.

**Pull request merged**